kernel/src/system/syscall.d

```diff
@@ -303,18 +303,44 @@ struct DirectoryListing {
 }
 
 @SyscallEntry(SyscallID.listDirectory)
-void listDirectory(void* listings_, size_t len) {
+void listDirectory(string path, void* listings_, size_t len, size_t start) {
 	import io.fs;
+	import kmain : rootFS;
 
 	DirectoryListing[] listings = (cast(DirectoryListing*)listings_)[0 .. len];
 
 	Process* process = getScheduler.currentProcess;
+	Node[] nodes;
+
+	if(path is null)
+	{
+		nodes = process.currentDirectory.nodes;
+	}
+	else
+	{
+		DirectoryNode node = cast(DirectoryNode)rootFS.root.findNode(path);
+		if (!node) {
+			process.syscallRegisters.rax = 0;
```

Comment:
IMO here it should return `-1` but afaik atm all syscalls return unsigned data types